localstack/utils/docker.py

```diff
@@ -1172,6 +1172,8 @@ def start_container(
                         if stdin:
                             sock.sendall(to_bytes(stdin))
                             sock.shutdown(socket.SHUT_WR)
+                        if not attach:
+                            return to_bytes(container.id), b""
```

Comment:
The problem I see with that, is that we break the compatibility with the docker cli client. For example this test:
```
    def test_start_with_interactive(self, docker_client: ContainerClient, create_container):
        container = create_container("alpine", command=["echo", "test_string"])
        stdout, _ = docker_client.start_container(container.container_id, interactive=True)
        assert stdout.decode(config.DEFAULT_ENCODING).strip() == container.container_id
```
Would fail on the CLI client and succeed with the SDK.
This was also the reason that interactive often also means attach, because the CLI does it. Features using this might then break, with the CLI client in use.

Reply:
Great catch, thanks @dfangl ! We should certainly make sure that the semantics are preserved across the two client implementations (SDK/CLI). 👍 

If we look at the documentation of the CLI client, it seems to reflect what we're doing in this PR. The `--attach` flag is used to attach to stdout/stderr, and `--interactive` is used to attach to stdin (allowing user input).
```
$ docker start --help

Usage:  docker start [OPTIONS] CONTAINER [CONTAINER...]

Start one or more stopped containers

Options:
  -a, --attach                  Attach STDOUT/STDERR and forward signals
      --checkpoint string       Restore from this checkpoint
      --checkpoint-dir string   Use a custom checkpoint storage directory
      --detach-keys string      Override the key sequence for detaching a container
  -i, --interactive             Attach container's STDIN

$ docker --version
Docker version 20.10.8, build 3967b7d
```

However - running a few quick tests, it seems that there is no (easy) way to attach stdin, while running the container in the background:

* Default (without `-i`/`-a` flags):
```
$ docker create --rm --name test alpine sh -c 'read in; echo $in; echo DONE'
...
$ docker start test
test
```

* With `-a` flag:
```
$ docker create --rm --name test alpine sh -c 'read in; echo $in; echo DONE' 
...
$ docker start test -a

DONE
```

* With `-i` flag:
```
$ docker create --rm --name test -i alpine sh -c 'read in; echo $in; echo DONE'
...
$ echo foobar | docker start test -i
foobar
DONE
```

--> As we can see, the last command correctly reads from stdin, but also attaches and prints the full stdout logs, so `-i` seems to imply `-t` (without having looked at the source code of the Docker CLI).

Let' discuss in more detail tomorrow. We may have to find an alternative solution to support the required use case...